packages/editor/src/components/post-actions/actions.js

```diff
@@ -470,7 +470,15 @@ const viewPostAction = {
 
 const postRevisionsAction = {
 	id: 'view-post-revisions',
-	label: __( 'View revisions' ),
+	getLabel( items ) {
```

Comment:
I didn't want to mix a prop with such different types, but don't have a strong opinion on this. A component could do that, but we use the label in specific internal components that render the actions and they expect a string returned value. 

Do you think it would be enough to add docs that we support a label component, but expect a string value?

Reply:
> they expect a string returned value.

What are these places?

Reply:
>they expect a string returned value.

Sorry, I should rephrase.. What I meant is that the label is used in places (menu items, tooltips, modal header title, etc..) that expect a final 'string' value - not per JS type per se. So while we can support components technically, personally I'd rather not do it from the start to try to keep consistency of the UI that is handled by the internal packages and not third party consumers. 

`__experimentalLabel` is equivalent yes, and also returns a `string` value in our codebase. If we returned something different - that we could - it would definitely create problems in UI. if you see most of the implementations of this callback it's either passed the block attributes, which is like our selected items or have a hook to get the current entity.

So, while I don't have a strong opinion to use a component, I'd rather we are just as flexible as we need right now. I don't think we have a use case right now for needing components, do we?